src/dune_rules/melange/melange_rules.ml

```diff
@@ -241,6 +241,64 @@ let setup_emit_cmj_rules ~sctx ~dir ~scope ~expander ~dir_contents
   in
   Buildable_rules.with_lib_deps ctx compile_info ~dir ~f
 
+module Runtime_deps = struct
+  let to_action_builder ~expander dep_conf =
+    let runtime_deps =
+      let runtime_deps, _sandbox = Dep_conf_eval.unnamed ~expander dep_conf in
+      runtime_deps
+    in
+    Action_builder.memoize "melange runtime deps" runtime_deps
+
+  let eval ~expander (deps : Dep_conf.t list) =
+    let open Memo.O in
+    let builder = to_action_builder ~expander deps in
+    let+ paths, _ = Action_builder.run builder Lazy in
+    paths
+
+  let targets ~output deps =
+    List.map
+      ~f:(fun src ->
+        let dst =
+          match output with
+          | `Private_library_or_emit output_dir ->
+            let target =
+              src |> Path.as_in_build_dir_exn
+              |> Path.Build.drop_build_context_exn
+            in
+            Path.Build.append_source output_dir target
+          | `Public_library (lib_dir, output_dir) ->
+            lib_output_path ~output_dir ~lib_dir src
+        in
+        (src, dst))
+      deps
+end
+
+let setup_runtime_assets_rules sctx ~dir ~target_dir ~mode
+    ~(mel : Melange_stanzas.Emit.t) ~output =
+  let open Memo.O in
+  let* melange_files =
```

Comment:
This naming got me confusing, the value contains a list of paths for the runtime deps right? What about `deps_paths`?

Reply:
good point, renamed in [b1df820](https://github.com/ocaml/dune/pull/7199/commits/b1df8200b43295b8f521e8f0a1b00ba3f7425f95)